Library/Homebrew/os/mac.rb

```diff
@@ -68,21 +68,7 @@ def languages
       end
       os_langs = os_langs.scan(/[^ \n"(),]+/)
 
-      @languages = [
-        *Homebrew.args.value("language")&.split(","),
-        *ENV["HOMEBREW_LANGUAGES"]&.split(","),
-        *os_langs,
-      ].uniq
-
-      # Ensure all languages are valid
-      @languages.select! do |lang|
-        Locale.parse(lang)
-        true
-      rescue Locale::ParserError
-        false
-      end
```

Comment:
If you want to remove this, you'll either need to translate legacy `AppleLangauges` values to modern locales ([it's a long list](https://github.com/apple/swift-corelibs-foundation/blob/42bfad499c4e938156d5b49331c75f69cc92798c/CoreFoundation/Locale.subproj/CFLocaleIdentifier.c#L345-L539)), or adjust the tests [which currently mandate that this returns valid locales](https://github.com/Homebrew/brew/blob/6b2dbbc96f7d8aa12f9b8c9c60107c9cc58befc4/Library/Homebrew/test/os/mac_spec.rb#L8).

Reply:
I changed the spec. Parsing to `Locale` is done inside the cask `DSL` anyways.